web/impact/impact/permissions/v1_api_permissions.py

```diff
@@ -1,12 +1,25 @@
 from accelerator_abstract.models.base_user_utils import is_employee
 from impact.permissions import (
     settings,
-    BasePermission)
+    BasePermission,
+)
+
+SAFE_METHODS = ['GET', 'HEAD', 'OPTIONS']
+
+
+def can_view_user_details_page(request):
+    # importing here to avoid circular import
+    from impact.v1.views.user_detail_view import UserDetailView  # noqa: E402
+    if request.resolver_match.url_name == UserDetailView.view_name:
```

Comment:
Because there is no else clause on this statement, this function will return a None (which is falsey) for all views that are not UserDetailView. This is probably not what you want. :)

Reply:
@jonkiparsky that is intentional, as I want the function to be default falsey in all the other views so that the permission check is determined by `is_employee(request.user)` and `request.user.groups.filter(name=settings.V1_API_GROUP).exists()` as `can_view_user_details_page(request)` will be False in those cases

Reply:
However, I like the idea of implementing it locally on the view

Reply:
yes, that's a very confusing pattern to me. let's bring it into the view